examples/demo-stf/src/sov-cli/main.rs

```diff
@@ -0,0 +1,200 @@
+        let priv_key = Self::generate();
+        let data = serde_json::to_string(&priv_key)?;
+        fs::create_dir_all(priv_key_path)?;
+        let path = Path::new(priv_key_path).join(format!("{}.json", priv_key.address));
+        fs::write(&path, data)?;
+        println!(
+            "private key written to path: {}",
+            path.into_os_string().into_string().unwrap()
+        );
+        Ok(())
+    }
+}
+
+impl SerializedTx {
+    fn new<P: AsRef<Path>>(
+        sender_priv_key_path: P,
+        module_name: &str,
+        call_data_path: P,
+        nonce: u64,
+    ) -> anyhow::Result<SerializedTx> {
+        let sender_priv_key = Self::deserialize_priv_key(sender_priv_key_path)?;
+        let sender_address = sender_priv_key.pub_key().to_address();
+        let message = Self::serialize_call_message(module_name, call_data_path)?;
+
+        let sig = Transaction::<C>::sign(&sender_priv_key, &message, nonce);
+        let tx = Transaction::<C>::new(message, sender_priv_key.pub_key(), sig, nonce);
+
+        Ok(SerializedTx {
+            raw: RawTx {
+                data: tx.try_to_vec()?,
+            },
+            sender: sender_address,
+        })
+    }
+
+    fn deserialize_priv_key<P: AsRef<Path>>(
+        sender_priv_key_path: P,
+    ) -> anyhow::Result<DefaultPrivateKey> {
+        let priv_key_data = std::fs::read_to_string(&sender_priv_key_path).with_context(|| {
+            format!(
+                "Failed to read private key from {:?}",
+                sender_priv_key_path.as_ref()
+            )
+        })?;
+
+        let sender_priv_key_data = serde_json::from_str::<PrivKeyAndAddress>(&priv_key_data)?;
+
+        Ok(DefaultPrivateKey::from_hex(
+            &sender_priv_key_data.hex_priv_key,
+        )?)
+    }
+
+    fn serialize_call_message<P: AsRef<Path>>(
+        module_name: &str,
+        call_data_path: P,
+    ) -> anyhow::Result<Vec<u8>> {
+        let call_data = std::fs::read_to_string(&call_data_path).with_context(|| {
+            format!(
+                "Failed to read call data from {:?}",
+                call_data_path.as_ref()
+            )
+        })?;
+        cmd_parser(&module_name, &call_data)
+    }
+}
+
+pub fn main() {
+    let cli = Cli::parse();
+
+    match cli.command {
+        Commands::SerializeCall {
+            sender_priv_key_path,
+            module_name,
+            call_data_path,
+            nonce,
+        } => {
+            let serialized =
+                SerializedTx::new(&sender_priv_key_path, &module_name, &call_data_path, nonce)
+                    .unwrap_or_else(|e| panic!("Call message serialization error: {}", e));
+
+            let mut bin_path = PathBuf::from(call_data_path);
+            bin_path.set_extension("dat");
+
+            let mut file = File::create(bin_path)
+                .unwrap_or_else(|e| panic!("Unable to crate .dat file: {}", e));
+
+            file.write_all(&vec![serialized.raw.data].try_to_vec().unwrap())
```

Comment:
We are writing the `borsh` data, should we convert it to hex? Can be done in a separate PR. What do you think @preston-evans98?

Reply:
I think we can make output format an option so they can specify bin, hex.. celestia-appd (and some other tools) expect it in hex, so that would definitely be convenient